src/main/webapp/app/tutor-course-dashboard/tutor-participation-graph/tutor-participation-graph.component.ts

```diff
@@ -42,6 +47,15 @@ export class TutorParticipationGraphComponent implements OnInit, OnChanges {
         if (this.numberOfParticipations !== 0) {
             this.percentageAssessmentProgress = Math.round((this.numberOfAssessments / this.numberOfParticipations) * 100);
         }
+        if (this.numberOfComplaints + this.numberOfMoreFeedbackRequests !== 0) {
+            this.percentageComplaintsProgress = Math.round(
+                ((this.numberOfComplaints -
+                this.numberOfOpenComplaints + // nr of evaluated complaints
+                    (this.numberOfMoreFeedbackRequests - this.numberOfOpenMoreFeedbackRequests)) / // nr of evaluated more feedback requests
+                    (this.numberOfComplaints + this.numberOfMoreFeedbackRequests)) * // total nr of complaints and feedback requests
+                    100,
+            );
+        }
```

Comment:
This is duplicated code. Also, onChanges also fires onInit, so you shouldn't need it here anyway.